catalog/core/catalog-core-subscriptionstore/subscriptionstore-impl/src/main/java/org/codice/ddf/catalog/subscriptionstore/common/SubscriptionPersistor.java

```diff
@@ -0,0 +1,171 @@
+                .collect(Collectors.toMap(SubscriptionMetadata::getId, Function.identity()));
+    }
+
+    /**
+     * Add the subscription metadata to the persistent store using it's id for the document key.
+     * <p>
+     * Insertion operations with the same key overwrite the previous value, which means a duplicate
+     * being added can be treated as a no-op, minus the network overhead.
+     *
+     * @param metadata the subscription metadata to add to the persistent store.
+     * @throws SubscriptionStoreException if a problem occurs during insert.
+     */
+    public void insert(SubscriptionMetadata metadata) {
+        LOGGER.debug("Adding [{}] to persistence store. ", metadata.getId());
+        PersistentItem persistentSubscription = metadataToPersistentItem(metadata);
+        try {
+            persistentStore.add(PersistentStore.EVENT_SUBSCRIPTIONS_TYPE, persistentSubscription);
+        } catch (PersistenceException e) {
+            throw new SubscriptionStoreException("Exception while persisting subscription: ", e);
+        }
+    }
+
+    /**
+     * Remove the subscription metadata from the persistent store that corresponds to the given id.
+     * <p>
+     * Redundant deletes are valid as a no-op to support javax.cache events propogating write-through
+     * processing to other cache nodes.
+     *
+     * @param subscriptionId the unique id of the subscription to delete.
+     * @throws SubscriptionStoreException if a problem occurs during delete.
+     */
+    public void delete(String subscriptionId) {
+        LOGGER.debug("Deleting [{}] from persistence store. ", subscriptionId);
+        try {
+            persistentStore.delete(PersistentStore.EVENT_SUBSCRIPTIONS_TYPE,
+                    getEcqlStringForId(subscriptionId));
+        } catch (PersistenceException e) {
+            throw new SubscriptionStoreException("Exception while deleting subscription: ", e);
+        }
+    }
+
+    /**
+     * Given subscription metadata, transform it into a persistent item for storage.
+     */
+    private PersistentItem metadataToPersistentItem(SubscriptionMetadata metadata) {
+        notNull(metadata, "subscription metadata cannot be null");
+
+        PersistentItem persistentItem = new PersistentItem();
+        persistentItem.addProperty(SUBSCRIPTION_ID_SOLR, metadata.getId());
+        persistentItem.addProperty(SUBSCRIPTION_TYPE, metadata.getType());
+        persistentItem.addProperty(SUBSCRIPTION_SERIALIZED_FILTER, metadata.getSerializedFilter());
+        persistentItem.addProperty(SUBSCRIPTION_CALLBACK_URL,
+                metadata.getCallbackAddress()
+                        .toString());
+
+        return persistentItem;
+    }
+
+    /**
+     * Given a result from the persistent store, transform the map back into valid subscription
+     * metadata.
+     */
+    private SubscriptionMetadata mapToMetadata(Map<String, Object> map) {
+        notEmpty(map, "persistentItem cannot be null or empty");
```

Comment:
Need to change this. There is a chance it will be a plain map if the persistent store is swapped out with a different implementation.